### PR TITLE
[JAX] Fix the wrong shape of bias when fusing GEMMs.

### PR DESCRIPTION
**Pull request by mingxu1067**
1. Fix the wrong shape of bias when fusing GEMMs.
2. Allow `update_collections` and `update_fp8_metas` to return both Dict and FrozenDict.

### CODE REVIEW

File: transformer_engine/jax/module.py
Comment (nouiz) [inline transformer_engine/jax/module.py:657]: from line 611:
```suggestion
                                                   self.bias_init, features,
```

File: transformer_engine/jax/fp8.py
Comment (nouiz) [inline transformer_engine/jax/fp8.py:204]: I didn't know about get_args.
Are you sure this syntax only support FrozenDict and Dict and not any other classes?
This syntax would make sure this doesn't cover any other classes and I think is easier to read:

```suggestion
        assert isinstance(original, (dict, FrozenDict))
        assert isinstance(new, (dict, FrozenDict))
```

File: transformer_engine/jax/module.py
Comment (mingxu1067) [reply transformer_engine/jax/module.py:429]: Done

File: transformer_engine/jax/module.py
Comment (mingxu1067) [reply transformer_engine/jax/module.py:657]: Done

File: transformer_engine/jax/fp8.py
Comment (mingxu1067) [reply transformer_engine/jax/fp8.py:204]: `(typing.Dict, <class 'flax.core.frozen_dict.FrozenDict'>)` is what get_args returned, which is equivalent to `(dict, FrozenDict)`.
The tought here is to reuse `Collection` (the datatype we defined in [fp8.py:21](https://github.com/NVIDIA/TransformerEngine/blob/2240c95bae734e9572b2023538938c9355240fac/transformer_engine/jax/fp8.py#L21)). And `get_args` is a standard approach of python to get type hints. Refer to [typing.html#typing.get_args](https://docs.python.org/3/library/typing.html#typing.get_args) for details.

Comment (nouiz) [reply transformer_engine/jax/fp8.py:204]: Maybe it is standard, but it is harder to understand. So fewer people will understand it at first read. This isn't great.
I don't block this PR for this. But readability is something useful.

I'll @timmoon10 finish the review/merge it.

Comment (mingxu1067) [reply transformer_engine/jax/fp8.py:204]: I agree this might make a bit worse readability, but helpful on scalability, in case that we would support more datatypes as `Collection`, such as `Collection = Union[Dict, FrozenDict, MyDefinedDict]`. Both ways we discussed are good to me. Please let me know the final decision, then I can make changes.

Comment (timmoon10) [reply transformer_engine/jax/fp8.py:204]: Let's do `(dict, FrozenDict)` for now. If we ever make `Collection` more complicated, then we should switch to `get_args(Collection)` at that point.

Comment (mingxu1067) [reply transformer_engine/jax/fp8.py]: Done

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**
